packages/provider/src/HocuspocusProvider.ts

```diff
@@ -414,6 +434,20 @@ export class HocuspocusProvider extends EventEmitter {
 
     this.removeAllListeners()
 
+    this.configuration.websocketProvider.off('connect', this.configuration.onConnect)
```

Comment:
My only concern is with unsubscribing from configuration methods. If the configuration is shared among multiple `HocuspocusProvider` instances, the handler will be removed for all of them when only one is destroyed. 

It seems like we could either automatically unsubscribe the configuration method when the configuration is no longer being used, or we could expose a method like `removeConfigurationListeners` and leave it up to the user. Open to your thoughts.

Reply:
Thanks for pointing this out! But indeed, as the configuration already contains the document identifier, it shouldn't be shared at all. Also I don't this we can work around this, so I think it's fine 👍